File: 7 kyu/Number of People in the Bus/solution.js
const number = (busStops) => busStops.reduce((rem, [on, off]) => rem + on - off, 0);
_____________________________________
var number = function(busStops){
  var totalPeople = 0;
  for (var i=0; i<busStops.length; i++) {
    totalPeople += busStops[i][0];
    totalPeople -= busStops[i][1];
  }
  return totalPeople;
}
_____________________________________
const number = busStops => busStops.reduce((p,n) => p+n[0]-n[1],0)
_____________________________________
var number = function(busStops){
  return busStops.map(x => x[0] - x[1]).reduce( (x, y) => x + y);
}

File: 7 kyu/Number of People in the Bus/solution.rb
def number(bus_stops)
  bus_stops.map {|(on,off)| on - off }.reduce(:+)
end
_____________________________________
def number(bus_stops)
  passengers = 0
  bus_stops.each do |a,b|
    passengers += a - b
  end
  passengers
end
_____________________________________
def number(bus_stops)
  riders = 0
  bus_stops.each do |stop|
    riders += stop[0]
    riders -= stop[1]
  end
  return riders
end
_____________________________________
def number(arr)
  arr.map{|a,b| a - b }.sum
end
_____________________________________
def number(bus_stops)
  bus_stops.reduce(0) { |k, (on, off)| k + on - off }
end
_____________________________________
def number(bus_stops)
  
  return bus_stops.reduce(0) { |sum,stop| sum + (stop[0] - stop[1]) }
  
end
_____________________________________
def number(bus_stops)
  bus_stops.map { | i | i.reduce(:-) }.reduce(:+)
end
_____________________________________
def number(bus_stops)
  bus_stops.inject(0) { |count, (p_in, p_out)| count + p_in - p_out }
end
